### PR TITLE
Stop leaking TypeScript library definitions through dependent packages

### PR DESCRIPTION
**Pull request by carlpaten**
I'm targeting Node.js 12.x, which means that I shouldn't be able to compile e.g. `"foo".replaceAll("a", "b")`. But if I depend on `ansi-codes`, *or any package that depends on it*, then that code compiles without error. This is a well-known issue, see e.g. https://github.com/microsoft/TypeScript/issues/33901.

This is caused by `ansi-code` dependency `type-fest` leaking library definitions. This is [fixed](https://github.com/sindresorhus/type-fest/pull/187) in the [latest version](https://github.com/sindresorhus/type-fest/releases/tag/v0.21.3).

This PR is part of a campaign to stop major packages from leaking TypeScript library definitions. If you merge it and cut a release I can visit packages that depend on `ansi-codes` and help them upgrade.

### CODE REVIEW

**Comment by sindresorhus** [comment]
https://github.com/sindresorhus/ansi-escapes/pull/24/checks#step:5:20

**Comment by carlpaten** [comment]
Node.js v8 didn't like the `tsd` upgrade. However it's been [EOL for more than a year](https://blog.risingstack.com/update-nodejs-8-end-of-life-no-support/). Would it be fine to remove it from the CI checks?

File: index.test-d.ts [hunk 1]
```diff
@@ -34,15 +34,15 @@ expectType<string>(ansiEscapes.clearScreen);
...
-expectType<string>(ansiEscapes.image(Buffer.from(1), {height: 1}));
-expectType<string>(ansiEscapes.image(Buffer.from(1), {height: '1px'}));
-expectType<string>(ansiEscapes.image(Buffer.from(1), {height: 'auto'}));
+expectType<string>(ansiEscapes.image(Buffer.from('1')));
+expectType<string>(ansiEscapes.image(Buffer.from('1'), {width: 1}));
+expectType<string>(ansiEscapes.image(Buffer.from('1'), {width: '1px'}));
+expectType<string>(ansiEscapes.image(Buffer.from('1'), {width: 'auto'}));
+expectType<string>(ansiEscapes.image(Buffer.from('1'), {height: 1}));
+expectType<string>(ansiEscapes.image(Buffer.from('1'), {height: '1px'}));
+expectType<string>(ansiEscapes.image(Buffer.from('1'), {height: 'auto'}));
```
Comment (Qix-) [inline index.test-d.ts:43]: These are not equivalent...

Comment (carlpaten) [reply index.test-d.ts:43]: I thought `tsd` only tested type definitions, and further I thought that `Buffer.from(1)` was just used to obtain an arbitrary `Buffer` instance.  So here we're obtaining an arbitrary buffer instance again. Did I err here?

Comment (Qix-) [reply index.test-d.ts:43]: Actually looking at this again, you're right. Idk what passing scalar `1` was originally intending to do but it is indeed equivalent looking at the docs.

Disregard

**Pull request merged**